### PR TITLE
Unit and integration tests

### LINKED ISSUE #230: [hacktoberfest] - Unit and integration tests
Test some important maths we have, like BitmapUtils methods

### PR DESCRIPTION
**Pull request by visheshjindal**
link #230 

## Description:
- Added mockk framework for the Mocking 
- Added few unit test cases for the BitmapUtils

## Check list for the Code Reviewer:
* [x] CHANGELOG
* [ ] README
* [ ] Wiki
* [x] Version Number

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [3.3.6] - 05/10/21
+### Added
+- Added Unit tests for BitmapUtils [#230](https://github.com/CanHub/Android-Image-Cropper/issues/230)
+
```
Comment (Canato) [inline CHANGELOG.md:15]: In this case we do not need changelog updates, cause this update don't affect the final user. 
Even that I think is one of the most important parts

File: cropper/src/test/java/com/canhub/cropper/BitmapUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    @After
+    fun teardown() {
+        unmockkObject(BitmapUtils)
+    }
+
+    @Test
+    fun `WHEN float array of rectangle points is provided, THEN result should be left value of the bounding rectangle`() {
+        // when
```
Comment (Canato) [inline cropper/src/test/java/com/canhub/cropper/BitmapUtilsTest.kt:27]: Let's please use upper case for this comments `// WHEN` I know this is not explained anywhere, sorry for this =/

File: cropper/src/test/java/com/canhub/cropper/BitmapUtilsTest.kt [hunk 3]
```diff
@@ -0,0 +1,126 @@
+package com.canhub.cropper
+
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+private val RECT_POINTS: FloatArray = floatArrayOf(1f,2f,3f,4f, 5f, 6f, 7f, 8f)
```
Comment (Canato) [inline cropper/src/test/java/com/canhub/cropper/BitmapUtilsTest.kt:10]: Let's extract the magic numbers into consts, like `rectLeft = 1f` so we can understand what which one represents

**Review by Canato — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (visheshjindal) [reply CHANGELOG.md:15]: Removing it

File: cropper/src/test/java/com/canhub/cropper/BitmapUtilsTest.kt (hunk 2 above)
Comment (visheshjindal) [reply cropper/src/test/java/com/canhub/cropper/BitmapUtilsTest.kt:27]: Okay

Comment (visheshjindal) [reply cropper/src/test/java/com/canhub/cropper/BitmapUtilsTest.kt]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (visheshjindal) [reply CHANGELOG.md]: Done

**Pull request merged**